### PR TITLE
chore: column_type_mappings

### PR DESCRIPTION
**Pull request by dungdm93**
Signed-off-by: Đặng Minh Dũng <dungdm93@live.com>

### SUMMARY
Cleanup column_type_mappings to correct & simplify type annotation
image

### TESTING INSTRUCTIONS
Run existing unit tests

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/base.py [hunk 1]
```diff
@@ -252,6 +256,11 @@ class BaseEngineSpec:  # pylint: disable=too-many-public-methods
             types.DateTime(),
             GenericDataType.TEMPORAL,
         ),
+        (
+            re.compile(r"^time", re.IGNORECASE),
+            types.Time(),
+            GenericDataType.TEMPORAL,
+        ),
```
Comment (villebro) [inline superset/db_engine_specs/base.py:263]: If placed here, it will match `timestamp`

**Review by villebro — COMMENTED** [review]
Thanks for the cleanup! Simplification looks good, but not sure if the moving `$time` in front of `$timestamp` will work

File: superset/db_engine_specs/base.py (hunk 1 above)
Comment (dungdm93) [reply superset/db_engine_specs/base.py]: Thanks. It's fixed

File: superset/db_engine_specs/base.py [hunk 2]
```diff
@@ -954,12 +952,12 @@ def select_star(  # pylint: disable=too-many-arguments,too-many-locals
 
         :param database: Database instance
         :param table_name: Table name, unquoted
-        :param engine: SqlALchemy Engine instance
+        :param engine: SqlAlchemy Engine instance
```
Comment (villebro) [inline superset/db_engine_specs/base.py:955]: nit: while we're fixing typos, let's get it _completely_ right 😆 
```suggestion
        :param engine: SQLAlchemy Engine instance
```

**Review by villebro — APPROVED** [review]
Thanks for the cleanup, this looks much better!

**Pull request merged**